Core/automation/lib/javascript/core/log.js

```diff
@@ -0,0 +1,208 @@
+                if (prefix != "log") {
+                    level = "[" + levelString + "] ";
+                    nameText = _name === null ? "" : _name;
+                }
+
+                if (prefix == "level") {
+                    return (level + msg.message);
+                }
+
+                var callerText;
+                if (msg.caller.substr(0,1) == "<") {
+                    callerText = ", in " + msg.caller;
+                } else {
+                    callerText = ", function " + msg.caller;
+                }
+
+                var message = msg.message == "" ? "" : msg.message + " ";
+
+                if (prefix == "short") {
+                    return (level + message + "\t\t[" + (nameText != "" ? nameText + ", " : "") + msg.fileName.split('/').pop() + ":" + msg.lineNumber + callerText + "]");
+                }
+                return (level + message + "\t\t[" + (nameText != "" ? nameText + ": " : "") + "at source " + msg.fileName.split(AUTOMATION_PATH).pop() + ", line " + msg.lineNumber + callerText + "]");
+            }
+
+            var _getCallerDetails = function _getCallerDetails (msg) {
+                var matches = msg.stack.split('\n\tat ')[3].match(/(.+?) \((.+):(\d+?)\)/);
+                msg.caller = matches[1];
+                msg.fileName = matches[2]
+                msg.lineNumber = matches[3]
+                return msg;
+            }
+
+            var _sendNotification = function _sendNotification (message, icon, levelString, log) {
+                if (_config[levelString].recipients !== undefined) {
+                    _config[levelString].recipients.forEach(function(mail){
+                        NotificationAction.sendNotification(mail, message, icon, levelString);
+                    })
+                    if (_config[levelString].recipients.length > 0) {
+                        log.trace(Error("Notification sent to " + _config[levelString].recipients.join(", ") + ". Message: \"" + message + "\""));
+                    }
+                } else {
+                    NotificationAction.sendBroadcastNotification(message, icon, levelString);
+                    log.trace(Error("Broadcast notification sent. Message: \"" + message + "\""));
+                }
+            }
+
+            var _legacyLoggerCorrection = function _legacyLoggerCorrection (msg) {
+                if (msg.fileName.search(/automation\/lib\/javascript\/core\/utils\.js$/) !== -1) {
+                    switch (msg.caller) {
+                        case "logError":
+                        case "logWarn":
+                        case "logInfo":
+                        case "logDebug":
+                        case "logTrace":
+                        case "error":
+                        case "warn":
+                        case "info":
+                        case "debug":
+                        case "log":
+                            var stackArray = msg.stack.split('\n\tat ');
```

Comment:
Can we use `/\s+at\s+/` rather than `'\n\tat '`?
The current version is very specific, and (for example) fails on GraalJS which uses spaces rather than tabs.

Reply:
I think we should be able to adjust it to also work with GraalJS. 
But we have to be very specific, as the stack also contains the message text, which can be whatever the user wants it to be, including multi-line.
Therefore I think your suggestions is way too unspecific, as it would grab occurrence of the word "at".
Worst case, I might need to identify the engine we are running on and define a regex for each specifically.

Reply:
Oh, good point. I just dropped that in to keep it working in my env. I would expect that ensuring it's the start of the line `/^\s+at\s+` would be better, or even just fixing it to tab OR 4x space `/^(\t|    )at\s+`. Personally I wouldn't try to detect the engine.